File: 2 Matrix Inversion/matrix_inversion.py
import numpy as np
from time import time
from numpy.linalg import inv

n = int(1e3)
trials = int(1e1)
run_time = []

for i in range(trials):
    A = np.random.randint(1, 100, (n, n))

    t1 = time()
    inv_A = inv(A)
    run_time.append(time() - t1)
    print(f"{i+1} runs complete")

print(f"Average runtime: {np.mean(run_time)}")  

File: 2 Matrix Inversion/matrix_inversion.m
clc; clear; close all;

n = 1e4;
trials = 1e1;
time = zeros(trials, 1);

for i = 1:trials
    A = ceil(100*rand(n, n));
    % must be invertable
    while(det(A) == 0)
        A = ceil(100*rand(n, n));
    end
    
    tic
    A_inv = inv(A);
    time(i) = toc;
    fprintf('%d runs complete\n', i)
end

fprintf('Average run time: %f seconds\n', mean(time))